newdle/client/src/util/hooks.js

```diff
@@ -0,0 +1,28 @@
+import {useState} from 'react';
+
+/**
+ * This hook handles the possible states regarding backend communication.
+ *
+ * @param {Function} func - the client function which will be invoked
+ */
+export function useClient(func) {
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState('');
```

Comment:
I think you could use a [`useReducer`](https://reactjs.org/docs/hooks-reference.html#usereducer) hook here to avoid having to do two separate updates